tools/ld-chroma-decoder/decoderpool.cpp

```diff
@@ -210,6 +210,18 @@ bool DecoderPool::putOutputFrame(qint32 frameNumber, const OutputFrame &outputFr
 
         // Save the frame data to the output file
         if (outputData.Y.size()) {
+            if (decoder.isOutputY4m()) {
```

Comment:
Can you write the header immediately after the file is opened in DecoderPool::process, rather than checking whether it's the first frame here? That'd also still include the header if no frames get output.

Reply:
Done.